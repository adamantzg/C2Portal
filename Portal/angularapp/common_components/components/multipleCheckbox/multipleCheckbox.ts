import { Component, Input } from '@angular/core';

@Component({
    selector: 'multiple-checkbox',
    templateUrl: './multipleCheckbox.html',
    styleUrls: ['./multipleCheckbox.css']
})
export class MultipleCheckboxComponent {
    @Input()
    displayField = 'name';
    @Input()
    idField = 'id';
    @Input()
    selectedField = 'selected';

    @Input()
    data: any[] = [];
}
