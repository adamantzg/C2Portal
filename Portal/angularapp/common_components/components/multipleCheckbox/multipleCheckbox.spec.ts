import { MultipleCheckboxComponent } from './multipleCheckbox';
import { ComponentFixture, async, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';

describe('GridComponent', () => {
    let component: MultipleCheckboxComponent;
    let fixture: ComponentFixture<MultipleCheckboxComponent>;

    setupTestBed({
        declarations: [MultipleCheckboxComponent],
        imports: [FormsModule]
    });

    beforeEach(() => {
      fixture = TestBed.createComponent(MultipleCheckboxComponent);
      component = fixture.componentInstance;

    });

    test('should create', () => {
      expect(component).toBeTruthy();
      expect(fixture).toMatchSnapshot();
    });
});
