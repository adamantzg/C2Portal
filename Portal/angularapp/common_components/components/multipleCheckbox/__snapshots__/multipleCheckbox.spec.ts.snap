// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridComponent should create 1`] = `
<multiple-checkbox
  data={[Function Array]}
  displayField={[Function String]}
  idField={[Function String]}
  selectedField={[Function String]}
>
  
</multiple-checkbox>
`;
